### ISSUE TITLE
Use native CUTEst and gcc if installed

### ISSUE THREAD

**Issue by abelsiqueira** [JuliaSmoothOptimizers/CUTEst.jl#58]
Improve `deps/build.jl` to reuse CUTEst and gcc if they are already installed.
See `MUMPS.jl` for example.

**Comment by dpo** [comment]
I wonder why this simple build script
```julia
using BinDeps
@BinDeps.setup
libcutest = library_dependency("libcutest", aliases=["libcutest_double"])
@BinDeps.install Dict(:libcutest => :libcutest)
```
fails to detect `libcutest`, which is in `/usr/local/lib` when installed with Homebrew... Would you mind trying it at your end? You can test with
```julia
julia> using BinDeps
julia> BinDeps.debug("CUTEst")
```

**Comment by dpo** [comment]
Closed by #85.
